datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -615,7 +611,8 @@ impl PruningPredicate {
         is_always_true(&self.predicate_expr) && self.literal_guarantees.is_empty()
     }
 
-    pub(crate) fn required_columns(&self) -> &RequiredColumns {
+    /// Return the columns that are required to evaluate the pruning predicate
+    pub fn required_columns(&self) -> &RequiredColumns {
```

Comment:
as it is in a different crate now this function needs to become pub, which I think is reasonable as it is part of the API

Likewise the change to make `RequiredColumns` pub